vagabond/gui/AddEntity.cpp

```diff
@@ -281,26 +283,38 @@ void AddEntity::searchPdb()
 
 void AddEntity::loadConfSpaceView(std::string suffix)
 {
-	ConfSpaceView *view = new ConfSpaceView(this, &_obj);
-	
-	if (suffix == "refined_torsions")
-	{
-		view->setMode(rope::ConfTorsions);
-	}
-	else if (suffix == "atom_positions")
-	{
-		view->setMode(rope::ConfPositional);
-	}
-
-	try
-	{
-		view->show();
-	}
-	catch (const std::runtime_error &err)
-	{
-		BadChoice *bc = new BadChoice(this, err.what());
-		setModal(bc);
-	}
+        try
+        {
+                if (_obj.modelCount()==0)
+                {
+                        throw std::invalid_argument("No models detected, have you modeled your entities?");
+                }
+
+                ConfSpaceView *view = new ConfSpaceView(this, &_obj);
+                if (suffix == "refined_torsions")
+                {
+                        view->setMode(rope::ConfTorsions);
+                }
+                else if (suffix == "atom_positions")
+                {
+                        view->setMode(rope::ConfPositional);
+                }
+                        try
+                {
+                        view->show();
+                }
+                catch (const std::runtime_error &err)
```

Comment:
See comment below: by catching `std::exception` you can remove this try/catch statement altogether and make it part of the bigger one.